saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SemanticRoleLabeling/SRLMultiGraphDataModel.scala

```diff
@@ -36,19 +39,27 @@ class SRLMultiGraphDataModel(parseViewName: String = null, frameManager: SRLFram
 
   val tokens = node[Constituent]((x: Constituent) => x.getTextAnnotation.getCorpusId + ":" + x.getTextAnnotation.getId + ":" + x.getSpan)
 
+  // Edges
+
   val sentencesToStringTree = edge(sentences, stringTree)
   val sentencesToTokens = edge(sentences, tokens)
   val sentencesToRelations = edge(sentences, relations)
   val relationsToPredicates = edge(relations, predicates)
   val relationsToArguments = edge(relations, arguments)
 
+  // Sensors
+
   sentencesToTokens.addSensor(CommonSensors.textAnnotationToTokens _)
   sentencesToRelations.addSensor(textAnnotationToRelation _)
-  sentencesToRelations.addSensor(textAnnotationToRelationMatch _)
+
+  // TODO - Verify if we really need this matching sensor. This is very inefficient.
+  //  sentencesToRelations.addSensor(textAnnotationToRelationMatch _)
```

Comment:
@bhargav we populate the relations two times, one with generating sensor that populates relations in the very beginning when we populate the sentences. Later we add candidate relations from XuePalmer candidate generator and we added matching sensor to make sure the generated Xue palmer candidates are also connected to the right sentences. I am not sure why it should work correctly without the matching sensor, especially in the case that we use constraints.